test/Timelock.t.sol

```diff
@@ -700,66 +852,439 @@ contract TimelockTest is Test {
+        targetAddresses[1] = address(lending);
+
+        bytes4[] memory selectors = new bytes4[](2);
+        selectors[0] = MockLending.deposit.selector;
+        selectors[1] = MockLending.withdraw.selector;
+
+        /// compare first 20 bytes
+        uint16[] memory startIndexes = new uint16[](2);
+        startIndexes[0] = 16;
+        startIndexes[1] = 16;
+
+        uint16[] memory endIndexes = new uint16[](2);
+        endIndexes[0] = 36;
+        endIndexes[1] = 36;
+
+        /// can only withdraw and deposit to timelock
+        bytes[] memory checkedCalldata = new bytes[](2);
+        checkedCalldata[0] = abi.encodePacked(address(timelock));
+        checkedCalldata[1] = abi.encodePacked(address(timelock));
+
+        bytes[] memory datas = new bytes[](1);
+        datas[0] = abi.encodeWithSelector(
+            timelock.addCalldataChecks.selector,
+            targetAddresses,
+            selectors,
+            startIndexes,
+            endIndexes,
+            checkedCalldata
+        );
+
+        vm.prank(address(safe));
+        timelock.scheduleBatch(
+            targets,
+            values,
+            datas,
+            bytes32(0),
+            MINIMUM_DELAY
+        );
+
+        bytes32 id = timelock.hashOperationBatch(
+            targets,
+            values,
+            datas,
+            bytes32(0)
+        );
+
+        assertTrue(
+            timelock.isOperationPending(id),
+            "operation should be pending"
+        );
+        assertTrue(timelock.isOperation(id), "operation should be present");
+        assertFalse(
+            timelock.isOperationReady(id),
+            "operation should not be ready"
+        );
+        assertFalse(
+            timelock.isOperationDone(id),
+            "operation should not be done"
+        );
+
+        assertEq(
+            timelock.getAllProposals()[0],
+            id,
+            "proposal should be in proposals"
+        );
+        assertEq(
+            timelock.getAllProposals().length,
+            1,
+            "proposal length incorrect"
+        );
+
+        vm.warp(block.timestamp + MINIMUM_DELAY);
+
+        vm.prank(address(safe));
+        timelock.executeBatch(targets, values, datas, bytes32(0));
+
+        address[] memory safeSigner = new address[](1);
+        safeSigner[0] = address(this);
+
+        safe.setOwners(safeSigner);
+
+        address[] memory lendingAddresses = new address[](2);
+        lendingAddresses[0] = address(lending);
+        lendingAddresses[1] = address(lending);
+
+        bytes[] memory lendingPayloads = new bytes[](2);
+        lendingPayloads[0] = abi.encodeWithSelector(
+            lending.deposit.selector,
+            address(timelock),
+            100
+        );
+        lendingPayloads[1] = abi.encodeWithSelector(
+            lending.withdraw.selector,
+            address(timelock),
+            100
+        );
+
+        timelock.executeWhitelistedBatch(
```

Comment:
check CallExecuted event emission

Reply:
I think we are missing even checks in other places as well, eg: [here](https://github.com/solidity-labs-io/safe-time-guard/blob/bb8bc2a3ef4b50ed8bde518ae3b901ed208252f0/test/Timelock.t.sol#L789). We can make a helper function that wraps `timelock.executeWhitelisted` and checks events within itself and also makes some post validations. When we call this helper function in tests events will be automatically validated. We can make similar helper methods for all the functions. I can make a pr on your pr if you like this idea. @ElliotFriedman